### PR TITLE
binfmt_misc daemonset

### PR DESCRIPTION
**Pull request by t0rr3sp3dr0**
This DaemonSet registers qemu-user-static binaries on binfmt_misc to allow multi-arch Docker images to be built.

### CODE REVIEW

File: dind/daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        app.kubernetes.io/name: kube-actions
+        app.kubernetes.io/component: binfmt-misc
+    spec:
+      initContainers:
+        - name: register
+          image: multiarch/qemu-user-static
+          args:
+            - --reset
+            - -p
+            - 'yes'
```
Comment (paulolieuthier) [inline dind/daemonset.yaml:23]: `--persistent` makes it clearer. :smile:

File: dind/daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: kubernetes.io/os
+                    operator: In
+                    values:
+                      - linux
+                  - key: kubernetes.io/arch
+                    operator: In
+                    values:
+                      - amd64
```
Comment (paulolieuthier) [inline dind/daemonset.yaml:49]: Is filterting by arch really necessary? I guess qemu will emulation will be useful if the hosts run on arm as well. qemu's setup will do the right thing either way: https://github.com/qemu/qemu/blob/master/scripts/qemu-binfmt-conf.sh#L292-L318.

**Review by paulolieuthier — APPROVED** [review]

File: dind/daemonset.yaml (hunk 1 above)
Comment (t0rr3sp3dr0) [reply dind/daemonset.yaml:23]: I'll change that!

File: dind/daemonset.yaml (hunk 2 above)
Comment (t0rr3sp3dr0) [reply dind/daemonset.yaml:49]: `qemu-static-user` is currently supported only on amd64 hosts: https://github.com/multiarch/qemu-user-static#supported-host-architectures

File: dind/daemonset.yaml (hunk 1 above)
Comment (t0rr3sp3dr0) [reply dind/daemonset.yaml:23]:
```suggestion
            - --persistent
            - 'yes'
```

**Pull request merged**